packages/genie/package.py

```diff
@@ -204,6 +191,13 @@ def setup_build_environment(self, spack_env):
         ):
             spack_env.prepend_path("ROOT_INCLUDE_PATH", str(self.spec[d.name].prefix.include))
 
+        # set compiler flags
+        cxxstd = self.spec.variants["cxxstd"].value
+        cxxstdflag = (
+            "" if cxxstd == "default" else getattr(self.compiler, "cxx{0}_flag".format(cxxstd))
+        )
+        spack_env.set("CXXFLAGS", f"-g -fno-omit-frame-pointer {cxxstdflag}")
+
```

Comment:
We try to avoid passing compilation flags through the environment if possible, and especially in `setup_build_environment()`. Please see the use of the optional method, `flag_handler()` in other autotools recipes such as `gettext()` The chained `ack` command `ack --python -g '/package\.py$' var/spack/repos/builtin/packages | ack -xl '\bdef\b[[:space:]]+flag_handler\b'` will give a full list of packages using that feature, and you can also examine the documentation for the default implementation in `lib/spack/spack/package.base.py`.

Reply:
ah great, i'll take a look at `flag_handler` and update this PR accordingly. thanks!

Reply:
updated the GENIE recipe to set C++ standard via the `flag_handler` method. let me know if you're happy with this change. thanks!